src/kv/untyped_map.h

```diff
@@ -152,36 +156,99 @@ namespace kv::untyped
           {
             // If we depend on the key existing, it must be present and have the
             // version that we expect.
-            if (!search.has_value() || (it->second != search.value().version))
+            if (
+              !search.has_value() ||
+              std::get<0>(it->second) != search.value().version ||
+              (track_conflicts &&
+               std::get<1>(it->second) != search.value().read_version))
             {
               LOG_DEBUG_FMT("Read depends on invalid version of entry");
               return false;
             }
+          }
+        }
+        return true;
+      }
+
+      void commit(Version v, kv::Version& max_conflict_version) override
+      {
+        auto& roll = map.get_roll();
+        auto current = roll.commits->get_tail();
+        auto state = current->state;
+        bool is_writing_to_new_key = false;
+        auto consensus = map.store->get_consensus();
+        bool track_commit = consensus != nullptr &&
```

Comment:
I am very uncomfortable with making commit() consensus-aware and dependent. Can we return the necessary version information to the caller instead? If not, can this at least be an optional callback provided by the caller?